metrics/metric.py

```diff
@@ -0,0 +1,159 @@
+            writer.write(image)
+        counter += 1
+
+
+def making_prediction(nr_images=-1):
+    prediction_dict = {}
+    counter = 0
+    img_sort_list = []
+    sorted_img_list = []
+
+    for pic in os.listdir(image_path)[0:nr_images+1]:
+
+        if pic == ".ipynb_checkpoints":
+            pass
+        else:
+            key = int(pic.split(".")[0])
+            key_img_tuple = (key, pic)
+            img_sort_list.append(key_img_tuple)
+
+    img_sort_list.sort(key=lambda x: x[0])
+
+    for pair in img_sort_list:
+        sorted_img_list.append(pair[1])
+
+    for img in sorted_img_list:
+        try:
+            prediction, scores, bboxes = predictor.predict(f"{image_path}/{img}")
+            prediction_dict[f"{counter}"] = prediction
+            counter += 1
+        except:
+            prediction_dict[f"{counter}"] = ["exception"]
+            counter += 1
+
+    return prediction_dict
+
+
+def calculating_score(nr_images=-1):
+    ingredients_dict = retrieve_info_from_csv(nr_images)
+    download_test_images(ingredients_dict) # passing nr_images is done for safety reasons only
+    prediction_dict = making_prediction(nr_images)  # passing nr_images is done for safety reasons only
+
+    total_amount_ingredients_counter = 0 #len([item[0] for item in ingredients_dict.values()])
+    total_amount_predictions_counter = 0
+    correct_predict_counter = 0
+    correct_predict_counter_oi = 0
+    false_predict_counter = 0
+    false_predict_counter_oi = 0
+
+    for key, value in ingredients_dict.items():
+        ingredients = value[0]
+        number_words = len(ingredients)
+        total_amount_ingredients_counter += number_words
+        predictions = prediction_dict[key]
+
+        for pred in predictions:
+            # iterate through every prediction and check if it is in ingredients
+            pred = pred.lower()
+            total_amount_predictions_counter += 1
+
+            if pred in ingredients:
+                correct_predict_counter += 1
+            else:
+                false_predict_counter += 1
+
+            for ingr in ingredients:
+                if (pred in oi_classes) and (pred == ingr):
+                    correct_predict_counter_oi += 1
+                else:
+                    false_predict_counter_oi += 1
+
+    print("##############################################")
+    print("Amount of correct predictions: ", correct_predict_counter)
+    print("Amount of false predictions: ", false_predict_counter)
+    print("Total amount of ingredients in csv file: ", total_amount_ingredients_counter)
+    print("Total amount of predictions made: ", total_amount_predictions_counter)
+    print("Amount of correct predictions found in algorithm classes: ",correct_predict_counter_oi)
+    #print("Amount of false predictions found in algorithm classes: ",false_predict_counter_oi)
+
+    accuracy = round(correct_predict_counter / total_amount_ingredients_counter, 2)
```

Comment:
I suggest to add another metric (accuracy ?) based on the correct_predict_counter_oi.